src/services/filesSharingServices.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * SPDX-FileCopyrightText: 2019 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+import axios from '@nextcloud/axios'
+import { generateOcsUrl } from '@nextcloud/router'
+
+import { SHARE } from '../constants.ts'
+import type {
+	createFileFromTemplateParams,
+	createFileFromTemplateResponse,
+	createFileShareParams,
+	createFileShareResponse,
+	getFileTemplatesListResponse,
+} from '../types/index.ts'
+
+/**
+ * Appends a file as a message to the messages list
+ * @param payload The function payload
+ * @param payload.path The file path from the user's root directory
+ * @param payload.shareWith The conversation's token
+ * @param payload.referenceId An optional reference id to recognize the message later
```

Comment:
```suggestion
 * @param payload.referenceId A reference id to recognize the message later
```
Everything is optional here